src/SessionManager/ProgramSettings.cc

```diff
@@ -34,6 +37,86 @@ void applyOptionalArgument(T& val, const string& argument_name, const cxxopts::P
 }
 
 ProgramSettings::ProgramSettings(int argc, char** argv) {
+    if (argc > 1) {
+        spdlog::info("Using command-line settings");
+    }
+    ApplyCommandLineSettings(argc, argv);
+
+    const fs::path user_settings_path = fs::path(getenv("HOME")) / CARTA_USER_FOLDER_PREFIX / "backend.json";
+    const fs::path system_settings_path = "/etc/carta/backend.json";
+
+    json settings;
+    if (fs::exists(system_settings_path) && !no_system_config) {
+        settings = JSONConfigSettings(system_settings_path.string());
+        system_settings_json_exists = true;
+        spdlog::info("Reading system settings from {}", system_settings_path.string());
+    }
+
+    if (fs::exists(user_settings_path) && !no_user_config) {
+        auto user_settings = JSONConfigSettings(user_settings_path.string());
+        user_settings_json_exists = true;
+        spdlog::info("Reading user settings from {}", user_settings_path.string());
+        settings.merge_patch(user_settings); // user on top of system
+    }
+
+    if (system_settings_json_exists || user_settings_json_exists) {
+        settings.merge_patch(command_line_settings); // force command-line on top of user and sytem
+        SetSettingsFromJSON(settings);
+    }
+}
+
+json ProgramSettings::JSONConfigSettings(const std::string& json_file_path) {
```

Comment:
I would suggest changing this function name to something like `JSONFromConfigFile` or `JSONSettingsFromFile`.